lib/view/weeklyTaskView.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:planner/common/database.dart';
+import 'package:planner/models/task.dart';
+import 'package:planner/view/taskView.dart';
+
+class WeeklyTaskView extends StatefulWidget {
+  const WeeklyTaskView({Key? key}) : super(key: key);
+
+  @override
+  _WeeklyTaskViewState createState() => _WeeklyTaskViewState();
+}
+
+class _WeeklyTaskViewState extends State<WeeklyTaskView> {
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+  DatabaseService _db = DatabaseService();
+  List<Task> _allTasks = [];
+  final GlobalKey<RefreshIndicatorState> _refreshIndicatorKey =
+      GlobalKey<RefreshIndicatorState>();
+  @override
+  void initState() {
+    super.initState();
+    fetchData();
+  }
+
+  Future<void> fetchData() async {
+    List<Task> tasks = await fetchWeeklyTask();
+    setState(() {
+      _allTasks = tasks;
+    });
+  }
+
+  Future<List<Task>> fetchWeeklyTask() async {
+    DateTime today = DateTime.now();
+    DateTime start = DateTime(today.year, today.month, today.day);
+    DateTime end = start.add(Duration(days: 7));
+
+    Map<DateTime, List<Task>> activeMap, delayedMap, completedMap;
+
+    // Fetch task maps for the specified week
+    (activeMap, delayedMap, completedMap) = await _db.getTaskMaps(start, end);
+
+    List<Task> allTasks = [
+      ...?activeMap[start],
+      ...?delayedMap[start],
+      ...?completedMap[start],
+    ];
+
+    print('All tasks for the week: $allTasks');
+
+    return allTasks;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    DateTime today = DateTime.now();
+    List<Widget> dayWidgets = [];
+
+    for (int i = 0; i < 7; i++) {
+      DateTime currentDate = today.add(Duration(days: i));
```

Comment:
I found this way of getting the next day breaks with stuff like daylight savings. I made a function for this in getDateOnly with an offset